anchore_engine/services/apiext/api/controllers/utils.py

```diff
@@ -302,48 +303,38 @@ def make_cvss_scores(metrics):
 
 
 def make_response_vulnerability_report(vulnerability_type, vulnerability_report):
-
     os_vulns = []
     non_os_vulns = []
 
-    vuln_matches = vulnerability_report.get("results", [])
-
-    for vuln_match in vuln_matches:
-        vulnerability_dict = vuln_match.get("vulnerability", {})
-        artifact_dict = vuln_match.get("artifact", {})
-        fix_dict = vuln_match.get("fix", {})
-        fix_versions = (
-            [item for item in fix_dict.get("versions", []) if item and item != "None"]
-            if fix_dict
-            else []
-        )
-        fixed_in_version = ", ".join(fix_versions) if fix_versions else "None"
+    # Convert the response from json to the model type
+    image_vulnerabilities = ImageVulnerabilitiesReport.from_json(vulnerability_report)
 
-        if not vulnerability_dict or not artifact_dict:
+    for result in image_vulnerabilities.results:
+        if not result.vulnerability or not result.artifact:
             logger.warn(
                 "Missing vulnerability and artifact data in match record, skipping"
             )
             continue
 
         record = {
-            "vuln": vulnerability_dict["vulnerability_id"],
-            "severity": vulnerability_dict["severity"],
-            "url": vulnerability_dict["link"],
-            "fix": fixed_in_version,
-            "package": "{}-{}".format(artifact_dict["name"], artifact_dict["version"]),
-            "package_name": artifact_dict["name"],
-            "package_version": artifact_dict["version"],
-            "package_type": artifact_dict["pkg_type"],
-            "package_cpe": artifact_dict["cpe"],
-            "package_cpe23": artifact_dict["cpe23"],
-            "package_path": artifact_dict["pkg_path"],
-            "feed": vulnerability_dict["feed"],
-            "feed_group": vulnerability_dict["feed_group"],
-            "nvd_data": vulnerability_dict["cvss_scores_nvd"],
-            "vendor_data": vulnerability_dict["cvss_scores_vendor"],
+            "vuln": result.vulnerability.vulnerability_id,
+            "severity": result.vulnerability.severity,
+            "url": result.vulnerability.link,
+            "fix": ",".join(result.fix.versions) if result.fix.versions else None,
```

Comment:
Just to be backwards compatible this needs to be the string "None"
```suggestion
            "fix": ",".join(result.fix.versions) if result.fix and result.fix.versions else "None",
```